core/corehttp/gateway_handler.go

```diff
@@ -76,80 +68,31 @@ func (i *gatewayHandler) loadTemplate() error {
 	return nil
 }
 
-func (i *gatewayHandler) resolveNamePath(ctx context.Context, p string) (string, error) {
-	p = gopath.Clean(p)
-
-	if strings.HasPrefix(p, IpnsPathPrefix) {
-		elements := strings.Split(p[len(IpnsPathPrefix):], "/")
-		hash := elements[0]
-		rp, err := i.node.Namesys.Resolve(ctx, hash)
-		if err != nil {
-			return "", err
-		}
-
-		elements = append(rp.Segments(), elements[1:]...)
-		p = gopath.Join(elements...)
-	}
-	if !strings.HasPrefix(p, IpfsPathPrefix) {
-		p = gopath.Join(IpfsPathPrefix, p)
-	}
-	return p, nil
-}
-
-func (i *gatewayHandler) ResolvePath(ctx context.Context, p string) (*dag.Node, string, error) {
-	p, err := i.resolveNamePath(ctx, p)
-	if err != nil {
-		return nil, "", err
-	}
-
-	node, err := i.node.Resolver.ResolvePath(ctx, path.Path(p))
-	if err != nil {
-		return nil, "", err
-	}
-	return node, p, err
-}
-
-func (i *gatewayHandler) NewDagFromReader(r io.Reader) (*dag.Node, error) {
+// TODO(cryptix):  find these helpers somewhere else
+func (i *gatewayHandler) newDagFromReader(r io.Reader) (*dag.Node, error) {
+	// TODO(cryptix): change and remove this helper once PR1136 is merged
+	// return ufs.AddFromReader(i.node, r.Body)
 	return importer.BuildDagFromReader(
 		r, i.node.DAG, i.node.Pinning.GetManual(), chunk.DefaultSplitter)
 }
 
-func NewDagEmptyDir() *dag.Node {
-	return &dag.Node{Data: ufs.FolderPBData()}
-}
-
-func (i *gatewayHandler) AddNodeToDAG(nd *dag.Node) (u.Key, error) {
-	return i.node.DAG.Add(nd)
-}
-
-func (i *gatewayHandler) NewDagReader(nd *dag.Node) (uio.ReadSeekCloser, error) {
-	return uio.NewDagReader(i.node.Context(), nd, i.node.DAG)
-}
-
-// TODO(btc): break this apart into separate handlers using a more expressive
-// muxer
+// TODO(btc): break this apart into separate handlers using a more expressive muxer
 func (i *gatewayHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	if i.config.Writable && r.Method == "POST" {
-		i.postHandler(w, r)
-		return
-	}
-
-	if i.config.Writable && r.Method == "PUT" {
-		i.putHandler(w, r)
-		return
-	}
-
-	if i.config.Writable && r.Method == "DELETE" {
-		i.deleteHandler(w, r)
-		return
-	}
-
-	if r.Method == "GET" {
-		i.getOrHeadHandler(w, r)
-		return
+	if i.config.Writable {
+		switch r.Method {
+		case "POST":
+			i.postHandler(w, r)
+			return
+		case "PUT":
+			i.putHandler(w, r)
+			return
+		case "DELETE":
+			i.deleteHandler(w, r)
+			return
+		}
 	}
 
-	if r.Method == "HEAD" {
+	if r.Method == "GET" || r.Method == "HEAD" {
```

Comment:
the above only is for `Writeable`. I didn't want to introduce another switch with just one case.